tests/e2e/nebulacluster/nebulacluster.go

```diff
@@ -0,0 +1,229 @@
+				StoragedReplicas: 3,
+			}, {
+				Name:             "test1-1-3",
+				GraphdReplicas:   1,
+				MetadReplicas:    1,
+				StoragedReplicas: 3,
+				Reference:        &kruiseReference,
+				SchedulerName:    nebulaSchedulerName,
+			}, {
+				Name:             "test1-3-3",
+				GraphdReplicas:   1,
+				MetadReplicas:    3,
+				StoragedReplicas: 3,
+				SchedulerName:    nebulaSchedulerName,
+			}, {
+				Name:             "test1-3-3",
+				GraphdReplicas:   1,
+				MetadReplicas:    3,
+				StoragedReplicas: 3,
+				Reference:        &kruiseReference,
+			}, {
+				Name:             "test2-3-4",
+				GraphdReplicas:   2,
+				MetadReplicas:    3,
+				StoragedReplicas: 4,
+				SchedulerName:    nebulaSchedulerName,
+			},
+		}
+
+		for i, tc := range testCases {
+			tc := tc
+			ginkgo.Context(fmt.Sprintf("%d: %#v", i, tc), func() {
+				ginkgo.It("should deploy, scale out, and scale in successfully", func() {
+					ginkgo.By("Deploy NebulaCluster")
+
+					var err error
+					// init the NebulaCluster Resource for testing
+					nc := getNebulaCluster(ns, tc.Name)
+					nc.Spec.Graphd.Replicas = pointer.Int32Ptr(tc.GraphdReplicas)
+					nc.Spec.Metad.Replicas = pointer.Int32Ptr(tc.MetadReplicas)
+					nc.Spec.Storaged.Replicas = pointer.Int32Ptr(tc.StoragedReplicas)
+					if tc.Reference != nil {
+						nc.Spec.Reference = *tc.Reference
+					}
+					if tc.SchedulerName != "" {
+						nc.Spec.SchedulerName = tc.SchedulerName
+					}
+
+					err = runtimeClient.Create(context.TODO(), nc)
+					framework.ExpectNoError(err, "failed to create NebulaCluster %s/%s", ns, nc.Name)
+
+					ginkgo.By("Wait for NebulaCluster ready")
+
+					err = waitForNebulaClusterReady(nc, 30*time.Minute, 30*time.Second, runtimeClient)
+					framework.ExpectNoError(err, "failed to wait for NebulaCluster %s/%s ready", ns, nc.Name)
+
+					ginkgo.By("Create port forward for NebulaCluster")
+
+					graphLocalAddress := "127.0.0.1"
+					var graphLocalPort int
+					var stopCh chan<- struct{}
+					graphLocalPort, stopCh, err = portForwardNebulaClusterGraphd(nc, graphLocalAddress, clientConfig)
+					framework.ExpectNoError(err, "failed to port forward for graphd of NebulaCluster %s/%s", ns, nc.Name)
+					defer close(stopCh)
+					framework.Logf("create port forward %s:%d for graphd of NebulaCluster %s/%s", graphLocalAddress, graphLocalPort, ns, nc.Name)
+
+					ginkgo.By("Init space and insert samples for NebulaCluster")
+					replicaFactor := 3
+					if tc.StoragedReplicas < 3 {
+						replicaFactor = 1
+					}
+					executeSchema := fmt.Sprintf(
+						"CREATE SPACE IF NOT EXISTS e2e_test(partition_num=%d,replica_factor=%d);",
+						5*tc.StoragedReplicas, replicaFactor) +
+						"USE e2e_test;" +
+						"CREATE TAG IF NOT EXISTS person(name string, age int);" +
+						"CREATE EDGE IF NOT EXISTS like(likeness double);"
+					err = waitForExecuteNebulaSchema(30*time.Second, 2*time.Second, graphLocalAddress, graphLocalPort, "user", "pass", executeSchema)
```

Comment:
If we run the e2e test without creating a new kind cluster, the graphLocalAddress and graphLocalPort will not work

Reply:
It dose work. And it have tested.
The network traffic forwards to `api-server`, and then to `graphd`. 
We can access the `api-server`, so the network traffic that forward to `graphd`.